src/Core/Managed/Shared/Extensibility/MetricAggregatorManager.cs

```diff
@@ -0,0 +1,275 @@
+        private static AggregatedMetricTelemetry CreateAggergatedMetricTelemetry(MetricAggregator aggregator, SimpleMetricStatisticsAggregator aggregatorStats)
+        {
+            if ((aggregator == null) || (aggregatorStats == null) || (aggregatorStats.Count <= 0))
+            {
+                return null;
+            }
+
+            var telemetry = new AggregatedMetricTelemetry(
+                aggregator.MetricName,
+                aggregatorStats.Count,
+                aggregatorStats.Sum,
+                aggregatorStats.Min,
+                aggregatorStats.Max,
+                aggregatorStats.StandardDeviation);
+
+            if (aggregator.Dimensions != null)
+            {
+                foreach (KeyValuePair<string, string> property in aggregator.Dimensions)
+                {
+                    telemetry.Properties.Add(property);
+                }
+            }
+
+            return telemetry;
+        }
+
+        /// <summary>
+        /// Takes snapshot of all active metric aggregators and turns results into metric telemetry.
+        /// </summary>
+        private void Snapshot()
+        {
+            ConcurrentDictionary<MetricAggregator, SimpleMetricStatisticsAggregator> aggregatorSnapshot =
+                Interlocked.Exchange(ref this.aggregatorDictionary, new ConcurrentDictionary<MetricAggregator, SimpleMetricStatisticsAggregator>());
+
+            // calculate aggregation interval duration interval
+            TimeSpan aggregationIntervalDuation = DateTimeOffset.UtcNow - this.lastSnapshotStartDateTime;
+            this.lastSnapshotStartDateTime = DateTimeOffset.UtcNow;
+
+            // prevent zero duration for interval
+            if (aggregationIntervalDuation.TotalMilliseconds < 1)
+            {
+                aggregationIntervalDuation = TimeSpan.FromMilliseconds(1);
+            }
+
+            // adjust interval duration to exactly snapshot frequency if it is close (within 1%)
+            double difference = Math.Abs(aggregationIntervalDuation.TotalMilliseconds - snapshotFrequency.TotalMilliseconds);
+
+            if (difference <= snapshotFrequency.TotalMilliseconds / 100)
+            {
+                aggregationIntervalDuation = snapshotFrequency;
+            }
+
+            if (aggregatorSnapshot.Count > 0)
+            {
+                foreach (KeyValuePair<MetricAggregator, SimpleMetricStatisticsAggregator> aggregatorWithStats in aggregatorSnapshot)
+                {
+                    AggregatedMetricTelemetry aggergatedMetricTelemetry = CreateAggergatedMetricTelemetry(aggregatorWithStats.Key, aggregatorWithStats.Value);
+
+                    aggergatedMetricTelemetry.Properties.Add(intervalDurationPropertyName, ((long)aggregationIntervalDuation.TotalMilliseconds).ToString(CultureInfo.InvariantCulture));
+
+                    if (aggergatedMetricTelemetry != null)
+                    {
+                        this.Client.Track(aggergatedMetricTelemetry);
```

Comment:
I just thought about it - I'd suggest to set an sdkVersion here different from just `dotnet: XX.XX.XX`. Something like `agg-net: XX.XXX.XX` so you'll now how many metrics were collected using this aggregation mechanism.